src/test/run-pass/issue-49955-2.rs

```diff
@@ -12,13 +12,15 @@
 
 use std::cell::Cell;
 
+const FIVE: Cell<i32> = Cell::new(5);
+
 #[inline(never)]
 fn tuple_field() -> &'static u32 {
     // This test is MIR-borrowck-only because the old borrowck
     // doesn't agree that borrows of "frozen" (i.e. without any
     // interior mutability) fields of non-frozen temporaries,
     // should be promoted, while MIR promotion does promote them.
-    &(Cell::new(5), 42).1
+    &(FIVE, 42).1
```

Comment:
calls to `Cell::new` are not promotable anymore, thus this breaks. The crater run didn't show any cases of people actually writing this kind of code (also needs MIR borrowck)